text/0089-scoped-api-keys.md

```diff
@@ -0,0 +1,128 @@
+- Title: Scoped API Keys
+- Start Date: 2021-10-15
+- Specification PR: [#89](https://github.com/meilisearch/specifications/pull/89)
+- Discovery Issue: [#51](https://github.com/meilisearch/product/issues/51)
+
+# Scoped API Keys
+
+## 1. Functional Specification
+
+### I. Summary
+
+The SDKs can generate `Scoped API Keys` inheriting from a MeiliSearch's `API key` generated to force filters during a search for an end-user. A `Scoped API Key` is generated on the user server-side code to be used by an end-user. It allows users to have multi-tenant indexes and thus restricts access to documents depending on the end-user making the search request.
```

Comment:
>  generated to force filters during a search for an end-user

This is confusing. What do you mean?

> A `Scoped API Key` is generated on the user server-side code

Do you mean the request to generate is made at the server side? Or is MeiliSearch instance the server side?

> depending on the end-user making the search request

I wouln't use `making the search request` as you may create a scoped API key for users managing indexes and not specifically for search requests.

Reply:
> generated to force filters during a search for an end-user

Q: This is confusing. What do you mean?
A: It means that a `Scoped API Key` is forged with a specific `filter` parameter that cannot be deleted. They are inserted in the `Scoped API Key` payload. When the client-code emits a search request, the search engine will first validate the authorization of the `Scoped API Key`, extract the Scoped API Key `filter` value and add the search request `filter` parameter behind an `AND` operator.

> A Scoped API Key is generated on the user server-side code

Q: Do you mean the request to generate is made at the server side? Or is MeiliSearch instance the server side?
A: It is generated by calling the SDK method on the server-side user code. MeiliSearch receives nothing and stores nothing when generating a Scoped API Keys when a user generates one for an end-user.

> depending on the end-user making the search request

Q: I wouldn't use making the search request as you may create a scoped API key for users managing indexes and not specifically for search requests.
A: For now, a `Scoped API Key` can only be used for a search request. Using a Scoped API Key to reach an endpoint other than /search will raise an `invalid_api_key` error. We can improve this in the future! 🤓